### ISSUE TITLE
Live test bug for test_disk_encryption_set_identity

### ISSUE THREAD

**Issue by PARADISSEEKR** [Azure/azure-cli#24689]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Related command**

live test for test_disk_encryption_set_identity

**To Reproduce**

`azdev test test_disk_encryption_set_identity --live`

debug: unrecognized arguments: --enable-purge-protection--retention-days

**Expected behavior**

**Environment summary**

```{
  "azure-cli": "2.42.0",
  "azure-cli-core": "2.42.0",
  "azure-cli-telemetry": "1.0.8",
  "extensions": {
  }
```

**Additional conext**

https://github.com/Azure/azure-cli/pull/23858/commits/b8c993756c7ce9f81205fb3dee18951397f6b091#diff-d47c29a68290175befec6271b7744d8f650a1245b8cc83f5ac0b45704fbd5d44

https://github.com/Azure/azure-cli/blob/c24160b0b8e651f6161aa6a85628e5d28ca1130e/src/azure-cli-testsdk/azure/cli/testsdk/preparers.py#L180-L181

Adding a space before `--retention-days 7` can temporarily solve this problem

**Comment by yonzhan** [comment]
Live test bug

**Comment by zhoxing-ms** [comment]
This issue was introduced by this PR #23858, @evelyn-ys could you please help take a look~

**Comment by evelyn-ys** [comment]
already fixed by #24636 

Thanks for reporting!
